sdk/monitor/azquery/custom_client.go

```diff
@@ -74,3 +74,56 @@ func (e *ErrorInfo) UnmarshalJSON(data []byte) error {
 func (e *ErrorInfo) Error() string {
 	return string(e.data)
 }
+
+// Row of data in a table, types of data used by service specified in LogsColumnType
+type Row []any
+
+// Table - Contains the columns and rows for one table in a query response.
+type Table struct {
+	// REQUIRED; The list of columns in this table.
+	Columns []*Column `json:"columns,omitempty"`
+
+	// REQUIRED; The name of the table.
+	Name *string `json:"name,omitempty"`
+
+	// REQUIRED; The resulting rows from this query.
+	Rows []Row `json:"rows,omitempty"`
+
+	// internal map used to lookup column index
+	columnIndexLookup map[string]int
+}
+
+// UnmarshalJSON implements the json.Unmarshaller interface for type Table.
+func (t *Table) UnmarshalJSON(data []byte) error {
+	var rawMsg map[string]json.RawMessage
+	if err := json.Unmarshal(data, &rawMsg); err != nil {
+		return fmt.Errorf("unmarshalling type %T: %v", t, err)
+	}
+	for key, val := range rawMsg {
+		var err error
+		switch key {
+		case "columns":
+			err = unpopulate(val, "Columns", &t.Columns)
+			delete(rawMsg, key)
+			t.columnIndexLookup = map[string]int{}
+			for i, v := range t.Columns {
+				t.columnIndexLookup[*v.Name] = i
+			}
+		case "name":
+			err = unpopulate(val, "Name", &t.Name)
+			delete(rawMsg, key)
+		case "rows":
+			err = unpopulate(val, "Rows", &t.Rows)
+			delete(rawMsg, key)
+		}
+		if err != nil {
+			return fmt.Errorf("unmarshalling type %T: %v", t, err)
+		}
+	}
+	return nil
+}
+
+// GetColumnIndex returns the index of the specified column
+func (t Table) GetColumnIndex(name string) int {
+	return t.columnIndexLookup[name]
```

Comment:
In the event that the value specified for `name` doesn't exist in `t.columnIndexLookup`, the result (i.e. `0`) is ambiguous

Reply:
fixed in commit [2613738](https://github.com/Azure/azure-sdk-for-go/pull/19374/commits/2613738f8d37eec9188058fac45247e1ee89a4cc)